src/main/java/org/generationcp/middleware/manager/LocationDataManagerImpl.java

```diff
@@ -142,14 +143,27 @@ public long countLocationsByCountry(final Country country) {
 
 	@Override
 	public List<Location> getFilteredLocations(final String programUUID, final Set<Integer> types, List<Integer> locationIds,
-		final List<String> locationAbbreviations, final boolean favourites) {
+		final List<String> locationAbbreviations, final boolean favourites, final String locationName, final Pageable pageable) {
 		if (!StringUtils.isEmpty(programUUID) && favourites) {
 			if (locationIds == null) {
 				locationIds = new ArrayList<>();
 			}
 			locationIds.addAll(this.getFavoriteProjectLocationIds(programUUID));
 		}
-		return this.daoFactory.getLocationDAO().filterLocations(programUUID, types, locationIds, locationAbbreviations);
+		return this.daoFactory.getLocationDAO().filterLocations(programUUID, types, locationIds, locationAbbreviations, locationName, pageable);
+	}
+
+	@Override
+	public long countFilteredLocations(final String programUUID, final Set<Integer> types,
+		List<Integer> locationIds,
+		final List<String> locationAbbreviations, final boolean favourites, final String locationName) {
+		if (!StringUtils.isEmpty(programUUID) && favourites) {
+			if (locationIds == null) {
+				locationIds = new ArrayList<>();
+			}
+			locationIds.addAll(this.getFavoriteProjectLocationIds(programUUID));
+		}
```

Comment:
Extract this common code:
`if (!StringUtils.isEmpty(programUUID) && favourites) {
			if (locationIds == null) {
				locationIds = new ArrayList<>();
			}
			locationIds.addAll(this.getFavoriteProjectLocationIds(programUUID));
		}`